src/hydrox/common/util.clj

```diff
@@ -46,3 +46,34 @@
      (-> proj
          (update-in [:source-paths] (fnil identity ["src"]))
          (update-in [:test-paths] (fnil identity ["test"]))))))
+
+(defn deps-read-project
+  "like `read-project` but for deps.edn'
+
+   (keys (deps-read-project (io/file \"example/deps.edn\")))
+   => (just [:description :license :name :source-paths :test-paths
+             :documentation :root :url :version :dependencies] :in-any-order)"
+  ([] (read-project (io/file "deps.edn")))
+  ([file]
+   (let [path (.getCanonicalPath file)
+         root  (subs path 0 (- (count path) 9))
+         url (str "github.com/parkside-securities/")
+         project-name (last (clojure.string/split root #"/"))
+         proj-map (read-string (slurp file))
+         dependencies (merge (:deps proj-map)
+                             (apply hash-map (remove nil? (map :extra-deps (vals (:aliases proj-map))))))
+         source-paths (vec (map (fnil identity ["src"]) (:paths proj-map)))
+         other-paths (vec
+                      (flatten
```

Comment:
With `mapcat`, we can eliminate the use of `flatten`. 
```clojure
(vec (remove nil? (mapcat :extra-paths (vals (:aliases proj-map)))))
```
I prefer threading macro it can be written as;
```clojure
(->> proj-map
         :aliases
         vals
         (mapcat :extra-paths)
         (remove nil?)
          vec)
```

Reply:
Here is another place where you could take advantage of `(concat coll nil)` which `mapcat` is using under the hood:

```clojure
(->> (:aliases proj-map)
     (vals)
     (mapcat :extra-paths)
     (vec))
```

Basically `concat` invoked with `nil` is a no-op due to Clojure testing each collection for [seq](https://github.com/clojure/clojure/blob/clojure-1.9.0/src/clj/clojure/core.clj#L718-L719)